### PR TITLE
Completed 5 game feature 5 upgrade and heal bug fixing

### PR DESCRIPTION
**Pull request by HYBleek**
I add more tests and fix some bug
TA's suggestion about my is open_close principle, so I arrange my class

### CODE REVIEW

File: src/main/java/usecase_essence_use/data_calculator/CalculatorCollectible.java [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-    int upgradeEssenceNeed;
 
-    public CalculatorCollectible(){
-        healEssenceRate = 1;
-        upgradeEssenceNeed =20;
-    }
+    public CalculatorCollectible(){}
+
+    public final int HP_ESSENCE_EXCHANGE_RATE = 1;
+    public final int ESSENCE_FOR_UPGRADE = 20;
```
Comment (stalight) [inline src/main/java/usecase_essence_use/data_calculator/CalculatorCollectible.java:8]: Naming convention is incorrect,
https://docs.oracle.com/javase/tutorial/java/nutsandbolts/variables.html
should just be hpEssenceExchangeRate and essenceForUpgrade

File: src/main/java/usecase_essence_use/data_calculator/CalculatorStat.java [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
+package usecase_essence_use.data_calculator;
 
 public class CalculatorStat {
-    int statIncrease;
+    public CalculatorStat(){}
 
-    public CalculatorStat(){
-        this.statIncrease = 20;
-    }
+    public final int STAT_INCREASE = 20;
```
Comment (stalight) [inline src/main/java/usecase_essence_use/data_calculator/CalculatorStat.java:6]: naming convention incorrect again, only capitalize and underscore if it is static final

File: src/main/java/usecase_essence_use/data_calculator/CollectibleUseManager.java [hunk 3]
```diff
@@ -45,6 +42,5 @@ public void spendCollectible(){
      */
```
Comment (stalight) [inline src/main/java/usecase_essence_use/data_calculator/CollectibleUseManager.java:42]: Parameter Num Essence has no description

**Review by stalight — CHANGES_REQUESTED** [review]

File: src/main/java/usecase_essence_use/data_calculator/CollectibleUseManager.java (hunk 3 above)
Comment (HYBleek) [reply src/main/java/usecase_essence_use/data_calculator/CollectibleUseManager.java:50]: Fixed

File: src/main/java/usecase_essence_use/data_calculator/CalculatorStat.java (hunk 2 above)
Comment (HYBleek) [reply src/main/java/usecase_essence_use/data_calculator/CalculatorStat.java]: fixed

File: src/main/java/usecase_essence_use/data_calculator/CalculatorCollectible.java (hunk 1 above)
Comment (HYBleek) [reply src/main/java/usecase_essence_use/data_calculator/CalculatorCollectible.java]: fixed

**Pull request merged**